src/merkle/store/mod.rs

```diff
@@ -373,6 +393,24 @@ impl MerkleStore {
 
         Ok(parent.into())
     }
+
+    // HELPER METHODS
+    // --------------------------------------------------------------------------------------------
+
+    /// Recursively clones a tree starting at the specified root into the specified target.
+    ///
+    /// If this Merkle store does not contain a tree with the specified root, this is a noop.
+    fn clone_tree_into(&self, root: RpoDigest, target: &mut Self) {
```

Comment:
Optional nit: the [std::clone::Clone::clone_from](https://doc.rust-lang.org/std/clone/trait.Clone.html#method.clone_from) implements the other way around: it takes self as mutable, and clones the data from a reference.

Reply:
I like this approach better. Changed to `clone_tree_from`.